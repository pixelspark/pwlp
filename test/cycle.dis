0000.	11	PUSHB	[03]
0002.	e1	USER	get_wall_time
0003.	21	PEEKB	1
0004.	82	BINARY	DIV
0005.	11	PUSHB	[08]
0007.	84	BINARY	MOD
0008.	11	PUSHB	[20]
0010.	83	BINARY	MUL
0011.	e1	USER	get_wall_time
0012.	22	PEEKB	2
0013.	82	BINARY	DIV
0014.	11	PUSHB	[03]
0016.	82	BINARY	DIV
0017.	11	PUSHB	[08]
0019.	84	BINARY	MOD
0020.	11	PUSHB	[20]
0022.	83	BINARY	MUL
0023.	e1	USER	get_wall_time
0024.	23	PEEKB	3
0025.	82	BINARY	DIV
0026.	11	PUSHB	[05]
0028.	82	BINARY	DIV
0029.	11	PUSHB	[08]
0031.	84	BINARY	MOD
0032.	11	PUSHB	[20]
0034.	83	BINARY	MUL
0035.	e0	USER	get_length
0036.	20	PEEKB	0
0037.	11	PUSHB	[01]
0039.	81	BINARY	SUB
0040.	24	PEEKB	4
0041.	74	UNARY	SHL8
0042.	86	BINARY	OR
0043.	23	PEEKB	3
0044.	74	UNARY	SHL8
0045.	74	UNARY	SHL8
0046.	86	BINARY	OR
0047.	22	PEEKB	2
0048.	74	UNARY	SHL8
0049.	74	UNARY	SHL8
0050.	74	UNARY	SHL8
0051.	86	BINARY	OR
0052.	e3	USER	set_pixel
0053.	01	POP	1
0054.	71	UNARY	DEC
0055.	60	JNZ	to 36
0058.	01	POP	1
0059.	e4	USER	blit
0060.	fe	SPECIAL	yield
0061.	03	POP	3
0062.	40	JMP	to 2
0065.	01	POP	1

