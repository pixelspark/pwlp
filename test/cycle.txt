// p = Cycle period
p = 3;
loop {
	r = (get_wall_time / p % 8) * 32;
	g = (get_wall_time / p / 3 % 8) * 32;
	b = (get_wall_time / p / 5 % 8) * 32;

	for(i = get_length) {
		set_pixel(i - 1, r, g, b);
	};

	blit;
	yield;
}