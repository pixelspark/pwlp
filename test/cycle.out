000000 000000 000000 000000 000000 000000 000000 000000 000000 000000 
e04020 e04020 e04020 e04020 e04020 e04020 e04020 e04020 e04020 e04020 
c0a060 c0a060 c0a060 c0a060 c0a060 c0a060 c0a060 c0a060 c0a060 c0a060 
c0e080 c0e080 c0e080 c0e080 c0e080 c0e080 c0e080 c0e080 c0e080 c0e080 
a020c0 a020c0 a020c0 a020c0 a020c0 a020c0 a020c0 a020c0 a020c0 a020c0 
8080e0 8080e0 8080e0 8080e0 8080e0 8080e0 8080e0 8080e0 8080e0 8080e0 
80c000 80c000 80c000 80c000 80c000 80c000 80c000 80c000 80c000 80c000 
602040 602040 602040 602040 602040 602040 602040 602040 602040 602040 
606060 606060 606060 606060 606060 606060 606060 606060 606060 606060 
40c0a0 40c0a0 40c0a0 40c0a0 40c0a0 40c0a0 40c0a0 40c0a0 40c0a0 40c0a0 
2000c0 2000c0 2000c0 2000c0 2000c0 2000c0 2000c0 2000c0 2000c0 2000c0 
